lib/workers/repository/changelog/index.ts

```diff
@@ -7,10 +7,10 @@ async function embedChangelog(upgrade): Promise<void> {
 }
 
 // istanbul ignore next
-export async function embedChangelogs(branches): Promise<void> {
+export async function embedChangelogs(branchUpgrades): Promise<void> {
   const upgrades = [];
-  for (const branch of branches) {
-    for (const upgrade of branch.upgrades) {
+  for (const branchName of Object.keys(branchUpgrades)) {
+    for (const upgrade of branchUpgrades[branchName]) {
```

Comment:
Unrelated to this pr, can you extract this please.

Reply:
It's not unrelated, all of the changes are required in order to solve #6337 - as discussed there, the key both needs to be allowed in the template, and the point at which the field is generated needs to be changed. Currently it's generated after the template is already compiled.

I'll create two new PRs for the separate pieces, but neither one will solve the issue on their own

Reply:
ok, sorry, i see it's because the changed input. You can leave it here.

Reply:
I've created https://github.com/renovatebot/renovate/pull/6346 and https://github.com/renovatebot/renovate/pull/6347

If that's what's preferred :)